net/Socket.cpp

```diff
@@ -540,7 +564,29 @@ int SocketPoll::poll(int64_t timeoutMaxMicroS)
         std::vector<CallbackFn> invoke;
         {
             std::lock_guard<std::mutex> lock(_mutex);
+            const size_t newConnCount = _newSockets.size();
+            const size_t globCount = GetStatsConnectionCount();
 
+            if( _limitedConnections &&
+                _connectionLimit > 0 &&
+                globCount + newConnCount > _connectionLimit)
+            {
+                // For now we simply drop new connections
+                const size_t overhead = globCount + newConnCount - _connectionLimit;
+                for(size_t i=0; i<overhead; ++i)
+                {
+                    std::shared_ptr<Socket>& socket = _newSockets.back(); // oldest
+                    assert(socket);
+
+                    LOG_WRN("Limiter: #" << socket->getFD() << ": Removing "
+                             << (i+1) << " / " << overhead << " new socket of (pre "
+                             << globCount << " + new " << newConnCount << ") / max " << _connectionLimit
+                             << " from " << _name);
+
+                    socket->resetThreadOwner();
```

Comment:
Interesting - looks like a useful feature; is there a test for this ? and do we build stats on / log how many dropped connections for admin-console / prometheus metrics ?

Reply:
tested in given unit tests.

no properly structured admin-console etc stats produced yet for this feature.